assembly/JSON.ts

```diff
@@ -298,77 +298,39 @@ export class Arr extends Value {
 
 export class Obj extends Value {
     _obj: Map<string, Value>;
-    keys: Array<string>;
 
     constructor() {
       super();
       this._obj = new Map();
-      this.keys = new Array();
     }
 
     toString(): string {
-      const objs: string[] = [];
-      for (let i: i32 = 0; i < this.keys.length; i++) {
-        let keyValueString = '"' + this.keys[i] + '": ';
-
-        // Cast our value into it's appropriate type
-        let value: Value | null = this._obj.get(this.keys[i]);
-        
-        // Check for null values
-        if (value == null || value.isNull) {
-          objs.push(keyValueString += "null");
-          continue;
-        }
-
-        // Cast to our proper type
-        if (value.isString) {
-          let castedValue = changetype<Str>(value);
-          keyValueString += '"' + castedValue.toString() + '"';
```

Comment:
And this didn't solve the problem? Could you expand on what use case needs the string quotes where it is missing? 🤔

Reply:
I'll address both your comments here.  The issue is that when we have a JSON Value and want to `stringify` it, `toString` it should produce a valid JSON string. According to the [spec](http://json.org/) JSON strings have quotations.  

So while this case solved it when generating an object's string value it doesn't fix it for arrays (see #227) and strings themselves.

Consider parsing the following string `"hello"`.  Since the parsing starts the string with `h` it doesn't know that it's parsing a string.  Wheras `` `"hello"` `` the first character is `"` which signals that the value is a string.

See: https://jsfiddle.net/93Lhfvu2/

As for how to get the real string value you added `valueOf` to access the internal string value.

Lastly I added the work of @kpeters-cbsi, which escapes internal `"`'s and other reserved characters.

Reply:
@willemneal 

> The issue is that when we have a JSON Value and want to stringify it, use toString, it should produce a valid JSON string. According to the spec JSON strings have quotations.

Ah yes! Thanks for providing the spec! Deifnitely when we output JSON, Strings should have quotation marks. But, it doesn't say anything about the `toString()` function having to output these quotation marks.

I guess taking a step back, I'm not debating whether if the strings shoud have quotation marks. More of a question if `toString()` is the right function for this to happen. Especially since [AssemblyScript's stdlib String toString()](https://www.assemblyscript.org/stdlib/string.html#instance-members) does not output quotation marks. It could be extremely confusing... 😯 

> So while this comment's case solved it when generating an object's string value it doesn't fix it for arrays (see #227) and strings themselves.

Ah! Okay cool, so this is more of an internal issue than! If it's arrays that are broken, I would say that we update how Arrays are stringified, and make sure to place the quotes around the strings. And/Or, we add a new function the JSON String like: `toFormatString()`, which includes the quotation marks, and update wherver we are stringifying a string to use this fucntion instead. Again, particularly because it could be really confusing given we already have a pattern in the AS stdlib for `toString()` 😄 

> Consider parsing the following string "hello". Since the parsing starts the string with h it doesn't know that it's parsing a string. Whereas `"hello"` the first character is " which signals that the value is a string.

Totally, for parsing JSON, we definitely want the quotes there, but if we do what I described above, we should add the quotations at the right time when we generate JSON 😄 

> As for how to get the real string value you added valueOf to access the internal string value.

Yep! I wanted them to be redundant. I was trying to create patterns here, where you could get the primitive type with "valueOf", and a string representation with "toString", which could then be used as normal. 

> Lastly I added the work of @kpeters-cbsi, which escapes internal "'s and other reserved characters.

That's awesome you included their work! Thank you! 😄 🎉